### ISSUE TITLE
NSUserDefaults vs CoreData

### ISSUE THREAD

**Issue by aaizuss** [uchicago-mobi/2016-Spring-Forum#87]
Just to clarify, in the instaWatchItOnTv project in the Spring Playground repo, you are using NSUserDefaults to save locally? And this is an alternative to using CoreData for local storage?

I know you discussed strategies in class, but can you remind me if there was any benefit to using CoreData over NSUserDefaults (besides getting more experience)? I was planning on using CoreData (because it seems like an important framework to get good at/confident with), but I'm more comfortable with using NSUserDefaults...

Also, I was using an embedded framework with files for the CoreDataStack (like last assignment) and for CloudKitManager. That makes sense, right? In your example you just have a DataKitManager file 

Also, are "iCloud Enabled Core Data Stack" slides from Lecture 4 relevant if I want to use the CoreData CloudKit combination, or was that for that thing you originally wanted us to do for the GameStore assignment that you decided was too hard...? (I'm confused because the slides never mention CloudKit, only iCloud - lecture 5 has the CloudKit stuff)

tldr I keep questioning everything I'm doing

**Comment by tabinks** [comment]
> I know you discussed strategies in class, but can you remind me if there was any benefit to using CoreData over NSUserDefaults (besides getting more experience)? I was planning on using CoreData (because it seems like an important framework to get good at/confident with), but I'm more comfortable with using NSUserDefaults...

Correct.  This is just showing how you could use `NSUserDefaults` to store some CloudKit data locally.  For this assignment, `NSUserDefaults` is not a good choice for local storage.  It is designed for small amounts of data, not images.  The main alternative to CoreData would be raw sqlite, or used custom objects you read/write to disk using `NSKeyedArchiver`.  CoreData would be the best, but it is not required.  Work with whichever technology you want to get more experience with.

> Also, I was using an embedded framework with files for the CoreDataStack (like last assignment) and for CloudKitManager. That makes sense, right? In your example you just have a DataKitManager file

Yes.  From now on you should put as much of your code in an embedded framework (for any project).  This will allow flexibility if you want to add extensions, watch apps, etc. to your project.   Also, with the dependency manager in Swift 3.0, this will allow you to reuse code easier.

> Also, are "iCloud Enabled Core Data Stack" slides from Lecture 4 relevant if I want to use the CoreData CloudKit combination, or was that for that thing you originally wanted us to do for the GameStore assignment that you decided was too hard...? (I'm confused because the slides never mention CloudKit, only iCloud - lecture 5 has the CloudKit stuff)

Thanks for paying attention.  No, iCloud Enable Core Data Stack is a different technology.  This syncs you CoreData store between devices without the need for a server.  It's a black box, but when it works its magical.  The downfall is the data is only for a single iCloud account.  So you could sync you own data back and forth between an iPad, iPhone and Mac app, but you could never share or access data from another user.  I'll run through this in a class so you can see the difference.  

> tldr I keep questioning everything I'm doing

At least you're asking the right questions.
